### PR TITLE
luca/db

### PR DESCRIPTION
**Pull request by luca020400**
- **Twelve: Add room**
- **Twelve: Playlist database**

### CODE REVIEW

**Comment by luca020400** [comment]
Note to self, need to zap indices on primary key tuple as that's done by default.

The item_id index is required in the crossref because the index is created on (playlist_id, item_d) which forces full table scans in case we search by item_id (like in associativity check)

File: app/src/main/java/org/lineageos/twelve/fragments/PlaylistsFragment.kt [hunk 1]
```diff
@@ -120,6 +159,32 @@ class PlaylistsFragment : Fragment(R.layout.fragment_playlists) {
         }
     }
 
+    private fun openCreateNewPlaylistDialog() {
+        val context = requireContext()
+
+        lateinit var editText: EditText
+
+        MaterialAlertDialogBuilder(context)
```
Comment (luca020400) [inline app/src/main/java/org/lineageos/twelve/fragments/PlaylistsFragment.kt]: Remember to add a style for rounded corner

File: app/src/main/java/org/lineageos/twelve/datasources/MediaDataSource.kt [hunk 2]
```diff
@@ -63,6 +63,30 @@ interface MediaDataSource {
 
     /**
      * Get the playlist information and all the tracks of the given playlist.
+     * If the playlist contains an audio that is unavailable, it will be mapped to null.
```
Comment (luca020400) [inline app/src/main/java/org/lineageos/twelve/datasources/MediaDataSource.kt:71]: Imo we should simply not map them in case.

**Comment by luca020400** [comment]
I wonder if it makes sense to have a more comprehensive insert cross ref method that also adds the item in case?

Imo we should just start looking at how to implement media mirroring to avoid (most) of this mess?

File: app/src/main/java/org/lineageos/twelve/datasources/MediaDataSource.kt (hunk 2 above)
Comment (SebaUbuntu) [reply app/src/main/java/org/lineageos/twelve/datasources/MediaDataSource.kt:71]: Users should have a chance to remove the entry even it it isn't available

File: app/src/main/java/org/lineageos/twelve/fragments/PlaylistsFragment.kt (hunk 1 above)
Comment (SebaUbuntu) [reply app/src/main/java/org/lineageos/twelve/fragments/PlaylistsFragment.kt]: Screenshot_20240922-123106
Looks like it only applies to PreferenceFragment

**Comment by luca020400** [comment]
Try a manual gradle debug build, you'll see some issues regarding transactions and suspend funcs, we should solve them

File: app/src/main/java/org/lineageos/twelve/database/entities/PlaylistWithBoolean.kt
Comment (luca020400) [inline app/src/main/java/org/lineageos/twelve/database/entities/PlaylistWithBoolean.kt:1]: migrate that to playlist commit

Comment (luca020400) [reply app/src/main/java/org/lineageos/twelve/database/entities/PlaylistWithBoolean.kt:1]: Also now with embedded we can remove toPair likely?

**Pull request merged**